editor/src/clj/editor/workspace.clj

```diff
@@ -364,11 +360,18 @@ ordinary paths."
            :header (format "The editor plugin '%s' is not compatible with this version of the editor. Please edit your project dependencies to refer to a suitable version." (resource/proj-path resource))}))
       false)))
 
+(defn- runduce!
+  "Like run!, but with added transducer"
+  [xform f coll]
+  (transduce xform (completing #(f %2)) nil coll)
+  nil)
+
 (defn- load-editor-plugins! [workspace added]
-  (let [added-resources (set (map resource/proj-path added))
-        plugin-resources (find-clojure-plugins workspace)
-        plugin-resources (filter (fn [x] (contains? added-resources (resource/proj-path x))) plugin-resources)]
-    (dorun (map (fn [x] (load-plugin! workspace x)) plugin-resources))))
+  (->> added
+       ;; FIXME: hack for extension-spine: spineguiext.clj requires spineext.clj
+       ;;        that need to be loaded first
+       (sort-by resource/proj-path util/natural-order)
+       (runduce! (filter is-plugin-clojure-file?) #(load-plugin! workspace %))))
```

Comment:
Performance improvement. The previous implementation went through all project resources, filtered them for Clojure files, then filtered once more to only include added files. There was no need to get all files since filtering changed files for Clojure files is enough.

Also, it's a miracle extension-spine worked with this code. It relied on resource list order to load `spineext.clj` before `spineguiext.clj`. For now, I leave the hack with an explicit description of the problem. A proper solution could be either:
- unpacking `editor/src` and adding it to the classpath, so the `require` of another plugin `.clj` file can work if it's not yet loaded;
- removing the `editor.spineext` require from `spineguiext.clj`.